src/Compilers/CSharp/Test/Syntax/Parsing/ParsingErrorRecoveryTests.cs

```diff
@@ -4562,8 +4570,124 @@ public void TestSemicolonAfterAnonymousTypeMemberComma()
             Assert.NotEqual(SyntaxKind.None, ds.Declaration.Variables[0].Initializer.EqualsToken.Kind());
             Assert.NotNull(ds.Declaration.Variables[0].Initializer.Value);
             Assert.Equal(SyntaxKind.AnonymousObjectCreationExpression, ds.Declaration.Variables[0].Initializer.Value.Kind());
-            Assert.Equal(1, file.Errors().Length);
-            Assert.Equal((int)ErrorCode.ERR_RbraceExpected, file.Errors()[0].Code);
+            Assert.Equal(3, file.Errors().Length);
+            Assert.Equal((int)ErrorCode.ERR_IdentifierExpected, file.Errors()[0].Code);
+            Assert.Equal((int)ErrorCode.ERR_SemicolonExpected, file.Errors()[1].Code);
+            Assert.Equal((int)ErrorCode.ERR_RbraceExpected, file.Errors()[2].Code);
```

Comment:
these all seem like unfortunate regressions.  this is def a tricky space.  i'm curious if we can do better with detecting if what comes should be eaten into the initializer, or should keep the old logic.